Update ee-install.rst (#3896)
mattermost/docs: source/install/ee-install.rst

 - Go to ``/etc/gitlab/gitlab.rb`` and set the following line to false

.. code-block:: text

    mattermost['enable'] = false

> - Run `sudo gitlab-ctl reconfigure` to apply the updated setting

Reviewer: This bullet point is misaligned compared to the bullet point above ("Go to /etc/gitlab/gitlab.rb and set the following line to false")

Author: Thanks, I've revised that and it's fixed. :)

Reviewer: It seems the bullet point is still misaligned here.

  * If you're using a modified version of Mattermost, using this tool will overwrite your changes and replace them with the official Enterprise Edition binary.
  * For versions prior to v5.27, please follow `these upgrade instructions <https://docs.mattermost.com/administration/upgrade.html#upgrading-team-edition-to-enterprise-edition>`_.

Navigate to **System Console > Edition and License** and select **Upgrade to Enterprise Edition**.

During the upgrade process, the Mattermost Enterprise Edition binary file that matches your current server version is downloaded, decompressed, and extracted. The Team Edition binary is then replaced by the Enterprise Edition version. Once this process is complete, you're prompted to restart your server.

The Mattermost version listed in **System Console > Edition and License** will change from **Team Edition** to **Enterprise Edition**, and you can now activate an Enterprise Edition trial or upload a license.

Permissions and Limitations
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> If you're using a package manager, such as GitLab Omnibus, to manage your Mattermost installation, the Mattermost system user won't have sufficient permissions to perform the upgrade. If this is the case, you'll need to change the file permissions manually.

Reviewer: ```suggestion
If you're using a package manager, such as GitLab Omnibus, to manage your Mattermost installation the Mattermost system user won't have sufficient permissions to perform the upgrade. If this is the case, you'll need to change the file permissions manually.
```

Changing the permissions in this way doesn't affect your Mattermost deployment or impact any data. The permission change is done solely for the upgrade.

To change the permissions using the command line on the Mattermost server, you need access to the command line tool as *mattermost* user. 

> Open the command line tool on the Mattermost server and ``cd`` to the Mattermost installation directory. Run the following commands (replacing ``<PathToBinaryFile>`` with the appropriate path (typically ``/opt/mattermost/bin/mattermost``) to change the ownership of the binary file to *mattermost* user and grant write access:

Reviewer: ```suggestion
Open the command line tool on the Mattermost server and ``cd`` to the Mattermost installation directory. Run the following commands (replacing ``<PathToBinaryFile>`` with the appropriate path (typically ``/opt/mattermost/bin/mattermost``)) to change the ownership of the binary file to *mattermost* user and grant write access:
```

  * If you're using a modified version of Mattermost, using this tool will overwrite your changes and replace them with the official Enterprise Edition binary.
  * For versions prior to v5.27, please follow `these upgrade instructions <https://docs.mattermost.com/administration/upgrade.html#upgrading-to-the-latest-version>`_.

Initiating the Conversion
~~~~~~~~~~~~~~~~~~~~~~~~~~

Navigate to **System Console > Edition and License** and select **Upgrade to Enterprise Edition**.

During the upgrade process, the Mattermost Enterprise Edition binary file that matches your version of Mattermost is downloaded, decompressed, extracted, and overwrites the Team Edition executable. Once this process is complete, you're prompted to restart your server. 

The Mattermost version listed in **System Console > Edition and License** will change from **Team Edition** to **Enterprise Edition**, and you can activate an E20 trial.

> GitLab Omnibus

Author: ```suggestion
Permissions and Limitations
```

~~~~~~~~~~~~~~

> If you’re using GitLab Omnibus to manage your Mattermost installation, you may encounter an error during the upgrade due to permission restrictions. This is resolved by changing the file permissions manually. Changing the permissions in this way doesn't affect your Mattermost deployment or impact any data. The permission change is done solely for the upgrade.

Author: ```suggestion
If you're using a package manager, such as GitLab Omnibus, to manage your Mattermost installation the Mattermost user won't have permissions to perform the upgrade. In this case you may encounter an error which is resolved by changing the file permissions manually. 

Changing the permissions in this way doesn't affect your Mattermost deployment or impact any data. The permission change is done solely for the upgrade.
```

> To change the permissions using the command line on the Mattermost server you need access to the command line tool as *mattermost* user.

Reviewer: ```suggestion
To change the permissions using the command line on the Mattermost server, you need access to the command line tool as *mattermost* user.
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you manually changed your file permissions, you can change them back:

1. Open the command line tool on the Mattermost server.
2. `cd` to the Mattermost installation directory.
3. Enter: 

.. code-block:: none
  
  \n\n```\nchown {{.FileUsername}} \"{{.Path}}\"\nchmod -w \"{{.Path}}\"\n```"

4. Press ENTER.

File permissions error
^^^^^^^^^^^^^^^^^^^^^^^

If your Mattermost deployment is part of a managed package you may receive file permissions errors and the upgrade will fail. You can edit the permissions settings manually:

1. Open the command line tool on the Mattermost server.
2. `cd` to the Mattermost installation directory.
3. Enter: 

.. code-block:: none

   \n\n```\nchown {{.MattermostUsername}} \"{{.Path}}\"\nchmod +w \"{{.Path}}\"\n```\n\

4. Press ENTER.

Incompatible system architecture
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This tool is currently only compatible with Linux systems using x86-64 architecture. If you’re running Mattermost on a different architecture, please follow the manual upgrade process.

Can’t get file
^^^^^^^^^^^^^^^^

> If the upgrade fails due to file retrieval failure, unavailable binary, or connectivity error please check your proxy settings and try again. If the problem persists, follow the manual upgrade process instead.

Reviewer: ```suggestion
If the upgrade fails due to file retrieval failure, unavailable binary, or connectivity error, please check your proxy settings and try again. If the problem persists, follow the manual upgrade process instead.
```

  * For versions prior to v5.27, please follow `these upgrade instructions <https://docs.mattermost.com/administration/upgrade.html#upgrading-to-the-latest-version>`_.

Navigate to **System Console > Edition and License** and select **Upgrade to Enterprise Edition**.

During the upgrade process, the Mattermost Enterprise Edition binary file that matches your version of Mattermost is downloaded, decompressed, extracted, and overwrites the Team Edition executable. Once this process is complete, you're prompted to restart your server. 

The Mattermost version listed in **System Console > Edition and License** will change from **Team Edition** to **Enterprise Edition**, and you can activate an E20 trial.

Permissions and Limitations
~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you're using a package manager, such as GitLab Omnibus, to manage your Mattermost installation the Mattermost user won't have permissions to perform the upgrade. In this case you may encounter an error which is resolved by changing the file permissions manually. 

Changing the permissions in this way doesn't affect your Mattermost deployment or impact any data. The permission change is done solely for the upgrade.

> To change the permissions using the command line on the Mattermost server, you need access to the command line tool as *mattermost* user. Open the command line tool on the Mattermost server, `cd` to the Mattermost installation directory, and run the following command.

Reviewer 2: @jespino I wrote that the typical path to the binary file is `/opt/mattermost/bin/mattermost`. Is that correct?

.. code-block:: none

>   \n\n```\nchown {{.MattermostUsername}} \"{{.Path}}\"\nchmod +w \"{{.Path}}\"\n```\n\

Reviewer 2: @jespino can you look at these commands and make sure they're correct? The commands from your PR had variable references to the system username and the path to the binary file. I believe that the system user needs to be `mattermost`, so I wrote that in.

.. code-block:: none

  \n\n```\nchown {{.FileUsername}} \"{{.Path}}\"\nchmod -w \"{{.Path}}\"\n```"

Note that any future automated updates or actions performed by other System Admins after the conversion will overwrite the conversion once the ``run gitlab-ctl configure`` command is run. This automatically updates Mattermost to the new version of Mattermost Team Edition, and overwrites Enterprise Edition. Any Enterprise Edition features you were using will no longer be accessible - but none of your user data will be affected.

Troubleshooting
~~~~~~~~~~~~~~~~

Mattermost has reverted to Team Edition
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you converted Team Edition to Enterprise Edition on a managed deployment and then upgraded, the upgrade will have overwritten Enterprise Edition with the latest version of Team Edition.

You can convert to Enterprise Edition again by following the steps above. If you plan to use Mattermost Enterprise Edition permanently, we recommend migrating your server to a self-managed one.

> The manual process reset my file permissions. How do I get them back?

Reviewer 2: ```suggestion
The manual process reset my file permissions
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you manually changed your file permissions, you can change them back.

1. Open the command line tool on the Mattermost server.
2. `cd` to the Mattermost installation directory.
3. Enter: 

.. code-block:: none
  

>   \n\n```\nchown {{.FileUsername}} \"{{.Path}}\"\nchmod -w \"{{.Path}}\"\n```"

Reviewer 2: ```suggestion
chown <OriginalFileOwner> <PathToBinaryFile>

chmod -w <PathToBinaryFile>
```

File permissions error
^^^^^^^^^^^^^^^^^^^^^^^

If your Mattermost deployment is part of a managed package you may receive file permissions errors and the upgrade will fail. You can edit the permissions settings manually:

1. Open the command line tool on the Mattermost server.
2. `cd` to the Mattermost installation directory.
3. Enter: 

.. code-block:: none

   \n\n```\nchown {{.MattermostUsername}} \"{{.Path}}\"\nchmod +w \"{{.Path}}\"\n```\n\

Incompatible system architecture
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This tool is currently only compatible with Linux systems using x86-64 architecture. If you’re running Mattermost on a different architecture, please follow the manual upgrade process.

> Can’t get file

Reviewer 2: ```suggestion
Can’t retrieve Enterprise Edition binary file
```

The Mattermost version listed in **System Console > Edition and License** will change from **Team Edition** to **Enterprise Edition**, and you can now activate an Enterprise Edition trial or upload a license.

Permissions and Limitations
~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you're using a package manager, such as GitLab Omnibus, to manage your Mattermost installation, the Mattermost system user won't have sufficient permissions to perform the upgrade. If this is the case, you'll need to change the file permissions manually.

Changing the permissions in this way doesn't affect your Mattermost deployment or impact any data. The permission change is done solely for the upgrade.

To change the permissions using the command line on the Mattermost server, you need access to the command line tool as *mattermost* user. Open the command line tool on the Mattermost server, `cd` to the Mattermost installation directory, and run the following command.

.. code-block:: none

  \n\n```\nchown {{.MattermostUsername}} \"{{.Path}}\"\nchmod +w \"{{.Path}}\"\n```\n\

> In the Mattermost System Console, retry the upgrade. When the upgrade is complete, return to the command prompt on the Mattermost server and run the following command to restore the file permissions, replacing <OriginalFileOwner> with the appropriate value:

Author: ```suggestion
In the Mattermost System Console, retry the upgrade. When the upgrade is complete, return to the command prompt on the Mattermost server and run the following command to restore the file permissions, replacing `<OriginalFileOwner>` with the appropriate value:
```
